Assets/SEE/XR/XRSEEActions.cs

```diff
@@ -0,0 +1,218 @@
+    {
+        hovering = true;
+        hoveredGameObject = args.interactableObject.transform.gameObject;
+        if (GlobalActionHistory.Current() == ActionStateTypes.Move)
+        {
+            oldParent = args.interactableObject.transform.parent;
+        }
+        hoveredGameObject.transform.TryGetComponent(out showLabel);
+        showLabel?.On();
+        if (hoveredGameObject.transform.TryGetComponent(out InteractableObject io))
+        {
+            io.SetHoverFlag(HoverFlag.World, true, true);
+        }
+    }
+    /// <summary>
+    /// This method gets called, when the user stops hovering over an interactable.
+    /// </summary>
+    /// <param name="args">Event data associated with the event when an Interactor stops hovering over an Interactable.</param>
+    public void OnHoverExited(HoverExitEventArgs args)
+    {
+        hovering = false;
+        hoveredGameObject = args.interactableObject.transform.gameObject;
+        hoveredGameObject.transform.TryGetComponent(out showLabel);
+        showLabel?.Off();
+        if (hoveredGameObject.transform.TryGetComponent(out InteractableObject io))
+        {
+            io.SetHoverFlag(HoverFlag.World, false, true);
+        }
+    }
+    /// <summary>
+    /// Is true, when the button for the primary actions is pressed.
+    /// </summary>
+    public static bool Selected { get; set; }
+
+    public static bool SelectedFlag { get; set; }
+    /// <summary>
+    /// The GameObject, which should be rotated.
+    /// </summary>
+    public static GameObject RotateObject { get; set; }
+    /// <summary>
+    /// This method gets called, when the button for the primary actions is pressed.
+    /// </summary>
+    /// <param name="context">Information provided to action callbacks about what triggered an action.</param>
+    private void Action(InputAction.CallbackContext context)
+    {
+        if (hovering)
+        {
+            if (GlobalActionHistory.Current() == ActionStateTypes.Move && Selected == true)
+            {
+                Selected = false;
+                SelectedFlag = false;
+            }
+            else
+            {
+                if (GlobalActionHistory.Current() == ActionStateTypes.Rotate)
+                {
+                    RayInteractor.TryGetCurrent3DRaycastHit(out RaycastHit hit);
+                    RotateObject = hit.collider.transform.gameObject;
+                    Selected = true;
+                }
+                Selected = true;
+                if (GlobalActionHistory.Current() != ActionStateTypes.Move)
+                {
+                    SelectedFlag = true;
+                }
+            }
+        }
+        else
+        {
+            InteractableObject.ReplaceSelection(null, true);
+        }
+    }
+    /// <summary>
+    /// Will be true, when the TreeView is open, while the user tries to move a node.
+    /// </summary>
+    public static bool CloseTreeView { get; set; }
```

Comment:
I'm a bit confused by the name of this field. As I understood it, it is true if the TreeView is open and the user currently tries to move a node. Then why is it called `CloseTreeView` (this would imply the opposite to me)?

Reply:
It is called that way, because when the TreeView is open and the user moves a node at the same time, the TreeView gets updated during the prozess, which is causing lags in VR. Therefore we close it beforehand to prevent the lag. I will update the documentation, so that it is more clear.